### ISSUE TITLE
Unable to reset password using "forgotten password"

### ISSUE THREAD

**Issue by vascovc** [pwm-project/pwm#50]
**What steps will reproduce the problem?**

1. Forgotten Password -> Input username
2. Receive email with code -> Input email code
3. Click check code

**What is the expected output? What do you see instead?**
Expected output => Option to set the new password
What do you see instead:
```
PWM 5015

An error has occurred. If this error occurs repeatedly please contact your helpdesk.
```

**What version of PWM are you using?**
1.8

**What ldap directory and version are you using?**
OpenLDAP: slapd 2.4.40

**Please paste any error log messages below:**
```
2016-02-12T18:21:40Z, TRACE, operations.PasswordUtility, {842} readPasswordPolicyForUser completed in 2ms [192.168.34.93]
2016-02-12T18:21:40Z, FATAL, servlet.AbstractPwmServlet, 5015 ERROR_UNKNOWN (unexpected error processing request: null
java.lang.NullPointerException
        at password.pwm.util.PwmPasswordRuleValidator.internalPwmPolicyValidator(PwmPasswordRuleValidator.java:222)
        at password.pwm.util.RandomPasswordGenerator.createRandomPassword(RandomPasswordGenerator.java:190)
        at password.pwm.ldap.auth.LDAPAuthenticationRequest.setTempUserPassword(LDAPAuthenticationRequest.java:406)
        at password.pwm.ldap.auth.LDAPAuthenticationRequest.authUsingUnknownPw(LDAPAuthenticationRequest.java:123)
        at password.pwm.ldap.auth.SessionAuthenticator.authUserWithUnknownPassword(SessionAuthenticator.java:170)
        at password.pwm.http.servlet.forgottenpw.ForgottenPasswordServlet.processSendNewPassword(ForgottenPasswordServlet.java:902)
        at password.pwm.http.servlet.forgottenpw.ForgottenPasswordServlet.advancedToNextStage(ForgottenPasswordServlet.java:761)
        at password.pwm.http.servlet.forgottenpw.ForgottenPasswordServlet.processAction(ForgottenPasswordServlet.java:236)
        at password.pwm.http.servlet.AbstractPwmServlet.handleRequest(AbstractPwmServlet.java:108)
        at password.pwm.http.servlet.AbstractPwmServlet.doPost(AbstractPwmServlet.java:64)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.AbstractPwmFilter$PwmFilterChain.doFilter(AbstractPwmFilter.java:110)
        at password.pwm.http.filter.CaptchaFilter.processFilter(CaptchaFilter.java:53)
        at password.pwm.http.filter.AbstractPwmFilter.doFilter(AbstractPwmFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.AbstractPwmFilter$PwmFilterChain.doFilter(AbstractPwmFilter.java:110)
        at password.pwm.http.filter.SessionFilter.processFilter(SessionFilter.java:81)
        at password.pwm.http.filter.AbstractPwmFilter.doFilter(AbstractPwmFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.AbstractPwmFilter$PwmFilterChain.doFilter(AbstractPwmFilter.java:110)
        at password.pwm.http.filter.ApplicationModeFilter.processFilter(ApplicationModeFilter.java:70)
        at password.pwm.http.filter.AbstractPwmFilter.doFilter(AbstractPwmFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.RequestInitializationFilter.doFilter(RequestInitializationFilter.java:95)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.GZIPFilter.doFilter(GZIPFilter.java:67)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.ApplicationStatusFilter.doFilter(ApplicationStatusFilter.java:36)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:521)
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1096)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:674)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1500)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1456)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:745)
)
2016-02-12T18:21:40Z, ERROR, http.PwmResponse, {842} 5015 ERROR_UNKNOWN (unexpected error processing request: null [192.168.34.93]
java.lang.NullPointerException
        at password.pwm.util.PwmPasswordRuleValidator.internalPwmPolicyValidator(PwmPasswordRuleValidator.java:222)
        at password.pwm.util.RandomPasswordGenerator.createRandomPassword(RandomPasswordGenerator.java:190)
        at password.pwm.ldap.auth.LDAPAuthenticationRequest.setTempUserPassword(LDAPAuthenticationRequest.java:406)
        at password.pwm.ldap.auth.LDAPAuthenticationRequest.authUsingUnknownPw(LDAPAuthenticationRequest.java:123)
        at password.pwm.ldap.auth.SessionAuthenticator.authUserWithUnknownPassword(SessionAuthenticator.java:170)
        at password.pwm.http.servlet.forgottenpw.ForgottenPasswordServlet.processSendNewPassword(ForgottenPasswordServlet.java:902)
        at password.pwm.http.servlet.forgottenpw.ForgottenPasswordServlet.advancedToNextStage(ForgottenPasswordServlet.java:761)
        at password.pwm.http.servlet.forgottenpw.ForgottenPasswordServlet.processAction(ForgottenPasswordServlet.java:236)
        at password.pwm.http.servlet.AbstractPwmServlet.handleRequest(AbstractPwmServlet.java:108)
        at password.pwm.http.servlet.AbstractPwmServlet.doPost(AbstractPwmServlet.java:64)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.AbstractPwmFilter$PwmFilterChain.doFilter(AbstractPwmFilter.java:110)
        at password.pwm.http.filter.CaptchaFilter.processFilter(CaptchaFilter.java:53)
        at password.pwm.http.filter.AbstractPwmFilter.doFilter(AbstractPwmFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.AbstractPwmFilter$PwmFilterChain.doFilter(AbstractPwmFilter.java:110)
        at password.pwm.http.filter.SessionFilter.processFilter(SessionFilter.java:81)
        at password.pwm.http.filter.AbstractPwmFilter.doFilter(AbstractPwmFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.AbstractPwmFilter$PwmFilterChain.doFilter(AbstractPwmFilter.java:110)
        at password.pwm.http.filter.ApplicationModeFilter.processFilter(ApplicationModeFilter.java:70)
        at password.pwm.http.filter.AbstractPwmFilter.doFilter(AbstractPwmFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.RequestInitializationFilter.doFilter(RequestInitializationFilter.java:95)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.GZIPFilter.doFilter(GZIPFilter.java:67)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.ApplicationStatusFilter.doFilter(ApplicationStatusFilter.java:36)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:521)
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1096)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:674)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1500)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1456)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:745)
)
2016-02-12T18:21:40Z, DEBUG, http.PwmResponse, {842} forcing logout due to error 5015 ERROR_UNKNOWN (unexpected error processing request: null [192.168.34.93]
java.lang.NullPointerException
        at password.pwm.util.PwmPasswordRuleValidator.internalPwmPolicyValidator(PwmPasswordRuleValidator.java:222)
        at password.pwm.util.RandomPasswordGenerator.createRandomPassword(RandomPasswordGenerator.java:190)
        at password.pwm.ldap.auth.LDAPAuthenticationRequest.setTempUserPassword(LDAPAuthenticationRequest.java:406)
        at password.pwm.ldap.auth.LDAPAuthenticationRequest.authUsingUnknownPw(LDAPAuthenticationRequest.java:123)
        at password.pwm.ldap.auth.SessionAuthenticator.authUserWithUnknownPassword(SessionAuthenticator.java:170)
        at password.pwm.http.servlet.forgottenpw.ForgottenPasswordServlet.processSendNewPassword(ForgottenPasswordServlet.java:902)
        at password.pwm.http.servlet.forgottenpw.ForgottenPasswordServlet.advancedToNextStage(ForgottenPasswordServlet.java:761)
        at password.pwm.http.servlet.forgottenpw.ForgottenPasswordServlet.processAction(ForgottenPasswordServlet.java:236)
        at password.pwm.http.servlet.AbstractPwmServlet.handleRequest(AbstractPwmServlet.java:108)
        at password.pwm.http.servlet.AbstractPwmServlet.doPost(AbstractPwmServlet.java:64)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:291)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.AbstractPwmFilter$PwmFilterChain.doFilter(AbstractPwmFilter.java:110)
        at password.pwm.http.filter.CaptchaFilter.processFilter(CaptchaFilter.java:53)
        at password.pwm.http.filter.AbstractPwmFilter.doFilter(AbstractPwmFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.AbstractPwmFilter$PwmFilterChain.doFilter(AbstractPwmFilter.java:110)
        at password.pwm.http.filter.SessionFilter.processFilter(SessionFilter.java:81)
        at password.pwm.http.filter.AbstractPwmFilter.doFilter(AbstractPwmFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.AbstractPwmFilter$PwmFilterChain.doFilter(AbstractPwmFilter.java:110)
        at password.pwm.http.filter.ApplicationModeFilter.processFilter(ApplicationModeFilter.java:70)
        at password.pwm.http.filter.AbstractPwmFilter.doFilter(AbstractPwmFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.RequestInitializationFilter.doFilter(RequestInitializationFilter.java:95)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.GZIPFilter.doFilter(GZIPFilter.java:67)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at password.pwm.http.filter.ApplicationStatusFilter.doFilter(ApplicationStatusFilter.java:36)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:521)
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1096)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:674)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1500)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1456)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:745)
)
2016-02-12T18:21:40Z, TRACE, http.PwmResponse, {842} forwarding to /WEB-INF/jsp/error.jsp [192.168.34.93]
2016-02-12T18:21:46Z, TRACE, http.PwmRequest, {842} GET request for: /pwm/ (no params) [192.168.34.93]
2016-02-12T18:21:46Z, TRACE, http.SessionManager, {842} incremented request counter to 1 [192.168.34.93]
2016-02-12T18:21:46Z, TRACE, http.PwmRequest, {842} GET request for: /pwm/private/ (no params) [192.168.34.93]
2016-02-12T18:21:46Z, TRACE, filter.AuthenticationFilter, {842} user requested resource requiring authentication (/pwm/private/), but is not authenticated; redirecting to LoginServlet [192.168.34.93]
2016-02-12T18:21:46Z, TRACE, http.PwmRequest, {842} GET request for: /pwm/private/login [192.168.34.93]
  posturl='H4sIAAAAAAAAAAFCAL3_UFdNLkFFUzEyOF9ITUFDMjU298-1mxKI8GKMqk3RMOCQ_4AWc16JqupX7dnNReKx7ZMvrlY2kexxQQXU4sepcUEg6fu3_UIAAAA='
2016-02-12T18:21:46Z, TRACE, http.PwmResponse, {842} forwarding to /WEB-INF/jsp/login.jsp [192.168.34.93]
2016-02-12T18:21:46Z, TRACE, http.SessionManager, {842} incremented request counter to 2 [192.168.34.93]

```

**Comment by clifford-github** [comment]
same Problem here. I'm using version

PWM v1.8.0-SNAPSHOT b11698842 r6aa2a6dd0d00ba946adc34f24e2bac86e0764340

**Comment by clifford-github** [comment]
Seems that this was fixed in commit 0c3e6f1c6ce6a94b508b7ab852a4f9bcea12fdca and merged into master in bf278d97d4f90ff63577c4e29703a6a462ae3942

I installed version

PWM v1.8.0-SNAPSHOT b12097071 r1947a7181a5fea79aaa75d30cac20454a8eac334

with

pwm-1.8.0-SNAPSHOT-2016-01-15T17:20:04Z-pwm-bundle.zip

and the option to set a new password is shown.

**Comment by jrivard** [comment]
Glad its working now, thanks for updating the bug!
